v3/spring-cloud-aws-v3-autoconfigure/src/main/java/io/awspring/cloud/v3/autoconfigure/secretsmanager/SecretsManagerConfigDataLocationResolver.java

```diff
@@ -82,22 +74,29 @@ public boolean isResolvable(ConfigDataLocationResolverContext context, ConfigDat
 			ConfigDataLocationResolverContext resolverContext, ConfigDataLocation location, Profiles profiles)
 			throws ConfigDataLocationNotFoundException {
 		registerBean(resolverContext, SecretsManagerProperties.class, loadProperties(resolverContext.getBinder()));
+		registerBean(resolverContext, CredentialsProperties.class,
+				loadCredentialsProperties(resolverContext.getBinder()));
 
 		registerAndPromoteBean(resolverContext, SecretsManagerClient.class, this::createAwsSecretsManagerClient);
 
-		SecretsManagerPropertySources propertySources = new SecretsManagerPropertySources(log);
+		SecretsManagerPropertySources propertySources = new SecretsManagerPropertySources();
 
 		List<String> contexts = getCustomContexts(location.getNonPrefixedValue(PREFIX));
 
 		List<SecretsManagerConfigDataResource> locations = new ArrayList<>();
 		contexts.forEach(propertySourceContext -> locations.add(
 				new SecretsManagerConfigDataResource(propertySourceContext, location.isOptional(), propertySources)));
 
+		if (!location.isOptional() && locations.isEmpty()) {
+			throw new SecretsManagerKeysMissingException(
+					"No Secrets Manager keys provided in `spring.config.import=aws-secretsmanager:` configuration.");
+		}
+
 		return locations;
 	}
 
 	private List<String> getCustomContexts(String keys) {
-		if (!StringUtils.isEmpty(keys)) {
+		if (StringUtils.hasLength(keys)) {
 			return Arrays.asList(keys.split(";"));
 		}
 		return Collections.emptyList();
```

Comment:
2 lines below on line 104 just add a comment. Can't pin point location since review can't be left on collapsed unmodified lines

`
/**
	 * Since hook can be activated more then one time, ApplicationContext needs to be
	 * checked if bean is already registered to prevent Exception. See issue #108 for more
	 * information.
	 */
`